package nl.han.oose.project.presentation.exception_mappers;
import jakarta.ws.rs.core.Response;
import nl.han.oose.project.business.exceptions.UserNotFoundException;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static java.net.HttpURLConnection.HTTP_NOT_FOUND;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class UserNotFoundExceptionMapperTest {
     private UserNotFoundExceptionMapper sut;
     private UserNotFoundException testException;
     private final String EXCEPTION_MESSAGE = "Test exception message";

     @BeforeEach
     public void setup() {
          this.sut = new UserNotFoundExceptionMapper();
          testException = new UserNotFoundException(EXCEPTION_MESSAGE);
     }

     @Test
     public void toResponseReturnsExpectedResponse() {
          Response response = this.sut.toResponse(testException);
          assertEquals(HTTP_NOT_FOUND, response.getStatus());
          assertEquals(EXCEPTION_MESSAGE, response.getEntity());
     }
}